### ISSUE TITLE
Hyphenated option breaks strict validation

### ISSUE THREAD

**Issue by samrg472** [yargs/yargs#1039]
Version: 10.0.3

Take this sample code:
```ts
import * as yargs from 'yargs';

yargs.option('awesome-arg', {
  string: true
}).strict().parse();

console.log('Parsed');
```

Running the sample with various arguments:
```
--awesomes-arg         // Strict mode is expected to fail here
--awesome-args         // and here
--awesomes-args        // here too
--awesomes-args --test // Strict mode only fails on `test`
--awesome-arg          // Strict mode successful as expected
```

**Comment by samrg472** [comment]
I managed to figure out what is happening.
```
if (/-/.test(key) && configuration['camel-case-expansion']) {
  addNewAlias(key, camelCase(key))
}
```
https://github.com/yargs/yargs-parser/blob/master/index.js#L350-L352

It looks like the `setArg` is only called on arguments from the command line. Does it make sense to add new aliases from that function? I tested camel case aliases without those 3 lines of code and camel case aliases still work as expected. This fixed the strict parser as well 😄

**Comment by bcoe** [comment]
@samrg472 let's start by adding a failing test case, if you don't mind 👍 then we can work on figuring out the best solution together (happy to help you land the work yourself).

CC: @porst17

**Comment by samrg472** [comment]
@bcoe Here's a test case, under the "strict mode" tests in the suite.
```
    it('fails when an invalid argument is provided', (done) => {
      yargs('--foo-bar')
        .strict()
        .fail((msg) => {
          return done()
        })
        .argv
    })
```

**Comment by bcoe** [comment]
@DennyGD @samrg472 is this still an issue, I know we've made some fixes around strict validation in recent months.

**Comment by samrg472** [comment]
@bcoe I ran the test again with version 13.2.2, the issue is still present. However, now the workaround provided above is no longer working. Any hyphen in the argument will bypass strict parsing for that particular argument.

**Comment by bcoe** [comment]
@samrg472 @jbottigliero I'm unable to recreate this issue, it seems like a key in config called `foo-bar` gets expanded to `fooBar` as expected.

**Comment by DennyGD** [comment]
@bcoe I can still reproduce it using yargs version `14.0.0` and node.js version `10.16.0`.

Sample code:

```
const yargs = require('yargs');

yargs
  .option('valid-option', {
    global: true,
    describe: 'Valid hyphenated option',
    type: 'boolean'
  })
  .command({
    command: 'test',
    describe: 'Test cmd',
    handler: (args) => console.log(JSON.stringify(args, null, 2))
  })
  .strict(true);

yargs.argv;

```

Run `node . test --valid-option` => command handler is executed (which is expected).


Run `node . test --unknown`. Output:
```
...
Unknown argument: unknown
```
Again, expected because we're in strict mode and option is not defined.


**And now the issue**.
Run `node . test --invalid-option`.
**Actual:** command handler is executed.
**Expected:** `Unknown argument: invalid-option` because this option is **not** defined and we're in strict mode.

**Comment by bcoe** [comment]
@DennyGD @samrg472, I believe this should finally have been addressed in `yargs@14.2.0`, thanks to the hard work of @andrewdotn.

Let us know if this does the trick for you.
